plugins/pluginmanagers/lz-n.nix

```diff
@@ -0,0 +1,130 @@
+{
+  lib,
+  helpers,
+  options,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+let
+  lzPluginType =
+    with helpers.nixvimTypes;
+    let
+      maybeRawStrOrList = oneOf [
+        str
+        (listOf (maybeRaw str))
+        rawLua
+      ];
+    in
+    submodule {
+      freeformType = attrsOf anything;
+      options = {
+        __unkeyed = mkOption {
+          type = str;
+          description = ''
+            The plugin's name (not the lua module name). This is what is passed to the load(name) function.
+            This is the name of the folder in /pack/*/opt/{name}.
+            See :h packadd
+          '';
+        };
+        enabled = helpers.defaultNullOpts.mkStrLuaFnOr bool true ''
+          When false, or if the function returns false, then this plugin will not be included in the spec.
+          This option corresponds to the `enabled` property of lz.n.
+        '';
+        beforeAll = helpers.mkNullOrLuaFn "Always executed before any plugins are loaded.";
+        before = helpers.mkNullOrLuaFn "Executed before this plugin is loaded.";
+        after = helpers.mkNullOrLuaFn "Executed after this plugin is loaded.";
+        event = helpers.mkNullOrOption maybeRawStrOrList "Lazy-load on event. Events can be specified as BufEnter or with a pattern like BufEnter *.lua";
+        cmd = helpers.mkNullOrOption maybeRawStrOrList "Lazy-load on command.";
+        ft = helpers.mkNullOrOption maybeRawStrOrList "Lazy-load on filetype.";
+        colorscheme = helpers.mkNullOrOption maybeRawStrOrList "Lazy-load on colorscheme.";
+        keys = helpers.mkNullOrOption' {
+          type = maybeRawStrOrList;
```

Comment:
This type being `listOf str` instead of `listOf anything` is the only reason your examples need to use raw lua. I don't see why users shouldn't be able to include attrs or lists in the keys list without resorting to raw lua?